### PR TITLE
[Review] JNI bindings for split

### PR DESCRIPTION
**Pull request by revans2**
Add in a java API for the split command.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 1]
```diff
@@ -1257,6 +1257,109 @@ public ColumnVector binaryOp(BinaryOp op, BinaryOperable rhs, DType outType) {
     }
   }
 
+  /**
+   * Splits a column (including null values) into a set of columns
+   * according to a set of indices. The caller owns the ColumnVectors and is responsible
+   * closing them
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]:
```suggestion
   * closing them.
```

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 2]
```diff
@@ -1257,6 +1257,109 @@ public ColumnVector binaryOp(BinaryOp op, BinaryOperable rhs, DType outType) {
...
+  public ColumnVector[] split(int... indices) {
+    try (ColumnVector cv = ColumnVector.fromInts(indices)) {
+      return split(cv);
+    }
+  }
+
+  /**
+   * Splits a column (including null values) into a set of columns
+   * according to a set of indices. The caller owns the ColumnVectors and is responsible
+   * closing them
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java]:
```suggestion
   * closing them.
```

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 3]
```diff
@@ -416,6 +416,43 @@ void testSliceWithArray() {
...
+          {28}};
+
+      ColumnVector[] splits = cv.split(1, 3, 5, 9);
+      try {
+        assertEquals(expectedData.length, splits.length);
+        for (int splitIndex = 0; splitIndex < splits.length; splitIndex++) {
+          ColumnVector subVec = splits[splitIndex];
+          subVec.ensureOnHost();
+          assertEquals(expectedData[splitIndex].length, subVec.getRowCount());
+          for (int subIndex = 0; subIndex < expectedData[splitIndex].length; subIndex++) {
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: Curious why assertColumnsAreEqual was not leveraged for this test and instead walks each result column directly.

Comment (revans2) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: I copied and pasted the test from `slice`.  If you want me to change it I am happy to.

Comment (jlowe) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java]: Oh, sorry, though this was a new test.  Would be nice to clean this up but not required.

**Comment by revans2** [comment]
@jlowe I have addressed your comments please take another look

**Comment by revans2** [comment]
rerun tests

**Pull request merged**